sonarr/episode.go

```diff
@@ -34,19 +34,56 @@ type Episode struct {
 	Series                   *Series        `json:"series"`
 }
 
+// GetMovie represents the input parameters for a movie api request.
+type GetEpisode struct {
+	// Set seriesID to get episodes for a specific series. Set to zero to get all episodes.
+	SeriesID int64
+	// Set seasonNumber to get episodes for a specific season. Set to zero to get all episodes.
+	SeasonNumber int
+	// Set episodeIds to get episodes for a specific set of ID's. Set to zero to get all episodes.
+	EpisodeIDS []int64
+	// Set episodeFileId to get episodes for a specific file. Set to zero to get all episodes.
+	EpisodeFileID int64
+	// Set includeImages to include images for each episode.
+	IncludeImages bool
+}
+
 // GetSeriesEpisodes returns all episodes for a series by series ID.
 // You can get series IDs from GetAllSeries() and GetSeries().
-func (s *Sonarr) GetSeriesEpisodes(seriesID int64) ([]*Episode, error) {
-	return s.GetSeriesEpisodesContext(context.Background(), seriesID)
+func (s *Sonarr) GetSeriesEpisodes(getEpisode *GetEpisode) ([]*Episode, error) {
+	return s.GetSeriesEpisodesContext(context.Background(), getEpisode)
 }
 
 // GetSeriesEpisodesContext returns all episodes for a series by series ID.
 // You can get series IDs from GetAllSeries() and GetSeries().
-func (s *Sonarr) GetSeriesEpisodesContext(ctx context.Context, seriesID int64) ([]*Episode, error) {
+func (s *Sonarr) GetSeriesEpisodesContext(ctx context.Context, getEpisode *GetEpisode) ([]*Episode, error) {
 	var output []*Episode
 
-	req := starr.Request{URI: bpEpisode, Query: make(url.Values)}
-	req.Query.Add("seriesId", fmt.Sprint(seriesID))
+	params := make(url.Values)
+
+	if getEpisode.SeriesID != 0 {
+		params.Set("seriesId", fmt.Sprint(getEpisode.SeriesID))
```

Comment:
I'm actually about to go through this entire library and remove all `fmt.Sprint` in favor of `strconv` methods. You could go ahead and change those here and now. I also did this: https://github.com/golift/starr/pull/138/files#diff-1e0a172dd940e7c7331f96972f1623dc1b9bb3a9c384c8f387d9a7ee1ebd31d8R312-R316

I'll go ahead and merge that so you can use it.